lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -464,9 +466,13 @@ def validate_selector(selector)
     end
 
     def validate_spec_with_kubectl(kubectl)
-      command = ["create", "-f", file_path, "--dry-run", "--output=name"]
-      _, err, st = kubectl.run(*command, log_failure: false, output_is_sensitive: sensitive_template_content?,
-        retry_whitelist: [:client_timeout], attempts: 3)
+      _, err, st = validate_with_dry_run_option(kubectl, "--dry-run")
+      if st.success? && sensitive_template_content?
```

Comment:
I don't think we should re-use `sensitive_template_content` I think we should add a new method / constant like `SERVER_DRY_RUNABLE`